nasa/ow_simulator
# Oceanwater 555 fix black image with lens flare

## Linked Issues:
| EPIC ⚡ | [OCEANWATER-86](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-86) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-555](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-555) |
| Github :octocat:  | # |


## Summary of Changes
* Subclassed gazebo's LensFlareSensorPlugin to allow color to be set in real-time.
* Set exact sun color (the brightness is the important part) using sunlight_values_for_shaders.py
* Specify compositor CameraLensFlare/PF_FLOAT32_R to be used by the plugin because it uses a floating point intermediate texture, necessary for our floating point shading pipeline.

## Test
* roslaunch ow europa_terminator.launch
* In rqt check checkboxes /ant_pan_position_controller/command, /ant_tilt_position_controller/command, and /StereoCamera/left/image_trigger
* Expand the pan and tilt topics to reveal their data settings.
* Set pan = 1.2 and tilt = -0.15
* Observe in rqt's Image View widget that you can see the terrain, the sun, and the lens flare all at the same time.
